sw/otbn/crypto/p256.s

```diff
@@ -15,9 +15,19 @@
 .globl proj_add
 .globl p256_generate_k
 .globl p256_generate_random_key
+.globl p256_key_from_seed
 
 .text
 
+/**
+ * Trigger a fault.
+ *
+ * P256 routines jump here if they suspect they are under fault attack and must
+ * abort the computation.
+ */
+p256_trigger_fault:
+  unimp
```

Comment:
It's a psuedoinstruction for a read from an invalid CSR, which triggers and `ILLEGAL_INSN` software error. OTBN software errors are recoverable alerts to the alert handler by default. OTBN can also be configured so that software errors are fatal, but it's all or nothing. The docs go into more detail here:
https://docs.opentitan.org/hw/ip/otbn/doc/#design-details-errors

However, I ended up having to change this part anyway because the branch to `unimp` broke the constant-time check. Technically, the branch does depend on the secret, it just depends specifically on whether the secret is 0 mod n, which would ~never (1/n \approx 2^-256 chance) happen under normal operation. So I changed this part of the code to be a subroutine that constructs a memory address based on the flag which is out of bounds (2^32 - 1) if the flag is set but 0 if it's not set, and loads a value from that address. If the flag is 1, it will produce the same kind of software error as `unimp`; if the flag is 0, it will harmlessly load and then discard a value.

Reply:
Created https://github.com/lowRISC/opentitan/issues/17305 to remind me to make a negative test for this functionality (this PR is already big enough that I don't want to just lump it in).